### DISCUSSION TITLE
Error when importing the logging module into ESP32 rover module

### DISCUSSION THREAD

**Discussion by obraganza** [micropython/micropython#15748] [ESP32]
I have been trying to import the logging module into my ESP32 rover module 
using  Thonny, however I always get an error as shown below.

`>>> import logging`
`Traceback (most recent call last):`
     `File "<stdin>", line 1, in <module>`
     `File "/lib/logging/__init__.py", line 618`
`SyntaxError: invalid syntax`

I have reloaded the library as well as the firmware and tried older versions of firmware getting similar results. 
Can someone advise me where the problem maybe.

Thanks
Oneil

**Comment by Tico06** [comment]
In Thonny, please double check the interpreter configured, in menu Run/Configure Interpreter.

**Comment by Josverl** [comment]
The Error `File "/lib/logging/__init__.py", line 618` is a strong indication that you are not importing the [MicroPython `logging.py` ](https://github.com/micropython/micropython-lib/blob/master/python-stdlib/logging/logging.py) module.
 - the line number is greater than the length of the micropython module
 -  the filename `__init__.py`  is different

I assume you have tried to install some other non-MicroPython logging module from somewhere , that does not work on MicroPython.
The logging module is a built-in module on most esp32 firmware.

Yoyu can check by running `help("modules")` on the micropython repl.

**Reply by obraganza** [reply]
Hi Josverl, 

you are right the logging module that I had used was incorrect.
I had used Thonny Tools (Package Manager) to search packages in PyPI and I ended up loading the wrong module.
I have now manually loaded the MicroPython logging module and it worked perfectly.
I could build a MicroPython program over the Arduino cloud using logging and other modules and it is working fine.

Thanks a lot for your help
